pysph/examples/surface_tension/capillary_wave.py

```diff
@@ -0,0 +1,257 @@
+        additional_props = ['V', 'color', 'scolor', 'cx', 'cy', 'cz',
+                            'cx2', 'cy2', 'cz2', 'nx', 'ny', 'nz', 'ddelta',
+                            'uhat', 'vhat', 'what', 'auhat', 'avhat', 'awhat',
+                            'ax', 'ay', 'az', 'wij', 'vmag2', 'N', 'wij_sum',
+                            'rho0', 'u0', 'v0', 'w0', 'x0', 'y0', 'z0',
+                            'kappa', 'arho', 'nu', 'wg', 'ug', 'vg',
+                            'pi00', 'pi01', 'pi02', 'pi10', 'pi11', 'pi12',
+                            'pi20', 'pi21', 'pi22']
+        gas = get_particle_array(
+            name='gas', x=fluid_x, y=fluid_y, h=h_fluid, m=m_fluid,
+            rho=rho_fluid, cs=cs_fluid, additional_props=additional_props)
+        gas.nu[:] = nu2
+        gas.color[:] = 0.0
+        liquid = get_particle_array(
+            name='liquid', x=circle_x, y=circle_y, h=h_circle, m=m_circle,
+            rho=rho_circle, cs=cs_circle, additional_props=additional_props)
+        liquid.nu[:] = nu1
+        liquid.color[:] = 1.0
+        wall = get_particle_array(
+            name='wall', x=wall_x, y=wall_y, h=h_wall, m=m_wall,
+            rho=rho_wall, cs=cs_wall, additional_props=additional_props)
+        wall.color[:] = 0.0
+        remove_overlap_particles(wall, liquid, dx_solid=dx, dim=2)
+        remove_overlap_particles(wall, gas, dx_solid=dx, dim=2)
+        remove_overlap_particles(gas, liquid, dx_solid=dx, dim=2)
+        gas.add_output_arrays(['V', 'color', 'cx', 'cy', 'nx', 'ny', 'ddelta',
+                               'kappa', 'N', 'scolor', 'p'])
+        liquid.add_output_arrays(['V', 'color', 'cx', 'cy', 'nx', 'ny',
+                                  'ddelta', 'kappa', 'N', 'scolor', 'p'])
+        wall.add_output_arrays(['V', 'color', 'cx', 'cy', 'nx', 'ny',
+                                'ddelta', 'kappa', 'N', 'scolor', 'p'])
+        for i in range(len(gas.x)):
+            R = sqrt(r(gas.x[i], gas.y[i]) + 0.0001*gas.h[i]*gas.h[i])
+            f = np.exp(-R/r0)
+            gas.u[i] = v0*gas.x[i]*(1.0-(gas.y[i]*gas.y[i])/(r0*R))*f/r0
+            gas.v[i] = -v0*gas.y[i]*(1.0-(gas.x[i]*gas.x[i])/(r0*R))*f/r0
+        for i in range(len(liquid.x)):
+            R = sqrt(r(liquid.x[i], liquid.y[i]) +
+                     0.0001*liquid.h[i]*liquid.h[i])
+            liquid.u[i] = v0*liquid.x[i] * \
+                (1.0 - (liquid.y[i]*liquid.y[i])/(r0*R))*f/r0
+            liquid.v[i] = -v0*liquid.y[i] * \
+                (1.0 - (liquid.x[i]*liquid.x[i])/(r0*R))*f/r0
+        return [liquid, gas, wall]
+
+    def create_solver(self):
+        kernel = QuinticSpline(dim=2)
+        integrator = PECIntegrator(liquid=TransportVelocityStep(),
+                                   gas=TransportVelocityStep())
+        solver = Solver(
+            kernel=kernel, dim=dim, integrator=integrator,
+            dt=dt, tf=tf, adaptive_timestep=False)
+        return solver
+
+    def create_equations(self):
```

Comment:
Can this not also use your `return_equations`?

Reply:
Sir, here the physical properties like c0, rho0 are different for different fluid arrays. I didn't put that functionality in the return_equations. Also, as this is working for only one scheme, I didn't make it use the return_equations.